src/Memoization.cpp

```diff
@@ -17,11 +17,17 @@ namespace Internal {
 namespace {
 
 class FindParameterDependencies : public IRGraphVisitor {
+    std::set<Function, Function::Compare> visited_functions;
+
 public:
     FindParameterDependencies() = default;
     ~FindParameterDependencies() override = default;
 
     void visit_function(const Function &function) {
+        if (visited_functions.count(function)) {
```

Comment:
As a stylistic choice, should we be always avoiding double-lookup? It's uglier to read but theoretically more efficient.
```
if (!visited_functions.insert(function).second) {
   return;
}
```

Reply:
I used structured bindings to make it a bit more readable IMO.